src/frontend/src/lib/types/token.ts

```diff
@@ -31,32 +31,41 @@ const TokenAppearanceSchema = z.object({
 	oisyName: TokenOisyNameSchema.optional()
 });
 
+// TODO: use Zod to validate the OnramperId
+const TokenBuySchema = z.object({
+	onramperId: z.custom<OnramperId>().optional()
+});
+
+const TokenBuyableSchema = z.object({
+	buy: z.custom<AtLeastOne<TokenBuy>>().optional()
+});
+
+const TokenSchema = z
+	.object({
+		id: TokenIdSchema,
+		network: NetworkSchema,
+		standard: TokenStandardSchema,
+		category: TokenCategorySchema
+	})
+	.merge(TokenMetadataSchema)
```

Comment:
Is `extend` closer to original definition?

Reply:
Unless I am mistaken (I haven't use `.merge` or `.extend` in Zod in a long time), to use `.extend`, I shall do

`.extend(TokenMetadataSchema.shape)`

Or not?

But if I look at `.merge` [documentation](https://zod.dev/?id=merge), it says that it is equivalent to it:

![Screenshot 2024-10-31 at 14 22 51](https://github.com/user-attachments/assets/9bb7767d-624c-4c02-95e2-cfd07bd972c2)


What is the most common way of doing it? WDYT?

Reply:
I don't think you need `.shape`, at least in the `oisy-wallet-signer` I just do extend.

https://github.com/dfinity/oisy-wallet-signer/blob/832a9d237109c99bb1fcec07e1b57c8457728eb3/src/types/rpc.ts#L36

But yeah if it's equivalent then all good.

Reply:
Ah maybe it's because in `oisy-wallet-signer` I extend with an object that contains the schema, gotcha.